pkg/dockertest/lakefs_container.go

```diff
@@ -0,0 +1,203 @@
+// Package dockertest provides a wrapper for running a lakeFS container in
+// integration tests.
+package dockertest
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/treeverse/lakefs/pkg/api/apigen"
+	"github.com/treeverse/lakefs/pkg/api/helpers"
+
+	"github.com/deepmap/oapi-codegen/pkg/securityprovider"
+	"github.com/ory/dockertest/v3"
+)
+
+const (
+	DockerLakeFSRepository  = "treeverse/lakefs"
+	ContainerTimeoutSeconds = 5 * 60
+	username                = "test user"
+)
+
+// waitForHealth waits until GETting URL returns 200.
+func waitForHealth(ctx context.Context, url string, pool *dockertest.Pool) error {
+	return pool.Retry(func() error {
+		req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+		if err != nil {
+			return err
+		}
+		res, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return err
+		} else if err = helpers.HTTPResponseAsError(res); err != nil {
+			return err
+		} else {
+			return nil
+		}
+	})
+}
+
+func New() *Build {
+	return &Build{}
+}
+
+// Build holds parameters for building a lakeFS dockertest container.
+type Build struct {
+	Repository string
+	// Tag is the docker tag to fetch.  It is required or New() will
+	// fail.
+	Tag string
```

Comment:
```suggestion
	DockerTag string
```
To avoid confusion with lakeFS tags